app/src/main/kotlin/at/bitfire/davdroid/sync/JtxSyncManager.kt

```diff
@@ -163,9 +165,14 @@ class JtxSyncManager @AssistedInject constructor(
 
         icalobjects.forEach { jtxICalObject ->
             // if the entry is a recurring entry (and therefore has a recurid)
-            // we udpate the existing (generated) entry
+            // we update the existing (generated) entry
             if(jtxICalObject.recurid != null) {
-                val local = localCollection.findRecurring(jtxICalObject.uid, jtxICalObject.recurid!!, jtxICalObject.dtstart!!)
+                val local = jtxICalObject.dtstart?.let { dtstart ->
+                    localCollection.findRecurring(jtxICalObject.uid, jtxICalObject.recurid!!, dtstart)
```

Comment:
It's not guaranteed that `jtxICalObject.recurid` is not null. We shouldn't use `!!`. Also the `let / ?: run` construct is a bit complex (and doesn't cover `jtxICalObject.recurid`). So I suggest to copy over the two variables:

~~~
val dtStart = …
val recurId = …
val local: LocalJtxObject
if (dtStart != null && recurId != null) {
  // it's now guaranteed, set local without !!
} else {
  logger.log(…)
  // null? See below
}
~~~

Reply:
makes sense

Reply:
although I find it highly unlikely that `jtxICalObject.recurid` will suddenly change to a null value if it was not at the time the if clause is evaluated